spec/api/api_put_invalid_spec.rb

```diff
@@ -0,0 +1,299 @@
+        end
+    end
+
+#!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    context 'verify user can update USER NAME' do
+        it 'verifies user can not update username with empty field' do
+            invalid_username = ''
+            #body = app_cl.generate_random_body
+            #app_cl.create_user(body)
+
+            body = app_cl.generate_random_body
+            response = app_cl.update_user(body[:username], app_cl.adjust_body(username: invalid_username))
+            expect(response.status).to eq(400)
+            user = app_cl.get_user(body[:username])
+            expect(user[:username]).to eq(body[:username])
+        end
+
+        it 'verifies user can not update username with negative values' do
+            invalid_username = -1
+            response = app_cl.update_user(body[:username], app_cl.adjust_body(username: invalid_username))
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can not update username with specials symbols' do
+            invalid_username = '@#-/'
+            response = app_cl.update_user(body[:username], app_cl.adjust_body(username: invalid_username))
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can not update username with mixed values' do
+            invalid_username = '@#-/Dark'
+            response = app_cl.update_user(body[:username], app_cl.adjust_body(username: invalid_username))
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can not update username with length 50 symbols' do
+            invalid_username = 'K' * 50
+            response = app_cl.update_user(body[:username], app_cl.adjust_body(username: invalid_username))
+            expect(response.status).to eq(400)
+        end
+    end
+
+
+    context 'verify user can update FIRST NAME' do
+        it 'verifies user can update with 1 symbol' do
+            body[:firstName] = 'F'
+            response = app_cl.update_user(body[:username], body)
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can update with empty field' do
+            body[:firstName] = ""
+            response = app_cl.update_user(body[:username], body)
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can update with specials symbols' do
+            body[:firstName] = '@/#$)'
+            response = app_cl.update_user(body[:username], body)
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can update with 50 letters' do
+            body[:firstName] = 'X' * 50
+            response = app_cl.update_user(body[:username], body)
+            expect(response.status).to eq(400)
+        end
+
+        it 'verifies user can update with numerical values' do
+            body[:firstName] = 123456
+            response = app_cl.update_user(body[:username], body)
+            expect(response.status).to eq(400)
+        end
+    end
```

Comment:
consider using shared examples or hashes to reduce code duplication

Reply:
working on optimization now